packages/measured-reporting/lib/reporters/Reporter.js

```diff
@@ -0,0 +1,169 @@
+const bunyan = require('bunyan');
+const Optional = require('optional-js');
+const { validateReporterParameters } = require('../validators/inputValidators');
+
+const DEFAULT_REPORTING_INTERVAL_IN_SECONDS = 10;
+
+/**
+ * The abstract reporter that specific implementations can extend to create a Self Reporting Metrics Registry Reporter.
+ *
+ * {@link SelfReportingMetricsRegistry}
+ * @abstract
+ */
+class Reporter {
+  /**
+   * @param {ReporterOptions} [options] The optional params to supply when creating a reporter.
+   */
+  constructor(options) {
+    if (this.constructor === Reporter) {
+      throw new TypeError("Can't instantiate abstract class!");
+    }
+
+    options = options || {};
+    validateReporterParameters(options);
+
+    /**
+     * Map of intervals to metric keys, this will be used to look up what metrics should be reported at a given interval.
+     * @type {Object.<number, Set<string>>}
+     * @private
+     */
+    this._intervalToMetric = {};
+    this._intervals = [];
+
+    /**
+     * Map of default dimensions, that should be sent with every metric
+     * @type {Dimensions}
+     * @protected
+     */
+    this._defaultDimensions = options.defaultDimensions || {};
+
+    /**
+     * Loggers to use, defaults to a new bunyan logger if nothing is supplied in options
+     * @type {Logger}
+     * @protected
+     */
+    this._log = options.logger || bunyan.createLogger({ name: 'Reporter', level: options.logLevel || 'info' });
+
+    /**
+     * the default interval a number in seconds.
```

Comment:
nitpick: maybe "the default reporting interval, a number in seconds", the comma helps

Reply:
- [ ] Update language here.